livre/core/pipeline/FilterInput.h

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2011-2014, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _FiterInput_h_
+#define _FiterInput_h_
+
+#include <livre/core/types.h>
+#include <livre/core/pipeline/InputPort.h>
+#include <livre/core/pipeline/PortData.h>
+
+namespace livre
+{
+
+/**
+ * The FilterInput class is used by the @PipeFilter class to instantiate
+ * the incoming connections to a given filter using the connection info
+ * provided by the filter.
+ *
+ * It provides functionality to wait on any or all the ports given by
+ * name.
+ *
+ * It also provides thread safe functions to query the state of the port.
+ */
+class FilterInput
+{
+public:
+
+    FilterInput();
+    ~FilterInput();
+
+    /**
+     * Gets the port according to given name.
+     * @param portName is the name of the port.
+     * @return the input port
+     * @throw std::runtime_error if the port with the port name
+     * is not available.
+     */
+    ConstInputPortPtr getPort( const std::string& portName ) const;
+
+    /**
+     * Queries if port with the given name is available.
+     * @param portName is the name of the port.
+     * @return true if there is a port with given name
+     */
+    bool hasPort( const std::string& portName ) const;
+
+    /**
+     * Waits for a given port name or for all the input ports if
+     * portname is empty.
+     * @param portName is the name of the port. Empty for all ports.
+     * @throw std::runtime_error if the port with the port name
+     * is not available.
+     */
+    void wait( const std::string& portName = "" ) const;
+
+    /**
+     * Waits for a any ready input to a given port ( The port
+     * can have many inputs )
+     * @param portName is the name of the port.
+     * @return true there are still waiting inputs.
+     * @throw std::runtime_error if the port with the port name
+     * is not available.
+     */
+    bool waitForAny( const std::string& portName ) const;
+
+    /**
+     * Waits for a any input to be ready.
+     * @param readyPorts The ready ports are listed into readyPorts array.
+     * @return true there are still waiting inputs.
+     */
+    bool waitForAny( lunchbox::Strings& readyPorts ) const;
```

Comment:
it will return true if you still need to wait for any port. 
If there are no ports connected, it will return false.
Or did I get your question right ? :)